### ISSUE TITLE
'ManyToOneRel' object has no attribute 'validators'

### ISSUE THREAD

**Issue by staubina** [inonit/drf-haystack#140]
When attempting to upgrade from `drf-haystack==1.6.1` to `drf-haystack==1.8.3` we are running into the following error.

```
E       AttributeError: 'ManyToOneRel' object has no attribute 'validators'
```
Is this a known issue when upgrading?  Did I by chance miss upgrading a dependency?

I have also upgraded Django, Django Rest Framework, and Haystack to the >= the required versions.

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../venvs/discovery/lib/python3.5/site-packages/rest_framework/serializers.py:563: in data
    ret = super(Serializer, self).data
../venvs/discovery/lib/python3.5/site-packages/rest_framework/serializers.py:262: in data
    self._data = self.to_representation(self.instance)
../venvs/discovery/lib/python3.5/site-packages/drf_haystack/serializers.py:226: in to_representation
    ret = super(HaystackSerializer, self).to_representation(instance)
../venvs/discovery/lib/python3.5/site-packages/rest_framework/serializers.py:513: in to_representation
    fields = self._readable_fields
../venvs/discovery/lib/python3.5/site-packages/django/utils/functional.py:35: in __get__
    res = instance.__dict__[self.name] = self.func(instance)
../venvs/discovery/lib/python3.5/site-packages/rest_framework/serializers.py:376: in _readable_fields
    field for field in self.fields.values()
../venvs/discovery/lib/python3.5/site-packages/rest_framework/serializers.py:363: in fields
    for key, value in self.get_fields().items():
../venvs/discovery/lib/python3.5/site-packages/drf_haystack/serializers.py:205: in get_fields
    kwargs = self._get_default_field_kwargs(model, field_type)
../venvs/discovery/lib/python3.5/site-packages/drf_haystack/serializers.py:133: in _get_default_field_kwargs
    kwargs.update(get_field_kwargs(field_name, model_field))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

field_name = 'course_runs', model_field = <ManyToOneRel: course_metadata.courserun>

    def get_field_kwargs(field_name, model_field):
        """
        Creates a default instance of a basic non-relational field.
        """
        kwargs = {}
>       validator_kwarg = list(model_field.validators)
E       AttributeError: 'ManyToOneRel' object has no attribute 'validators'

../venvs/discovery/lib/python3.5/site-packages/rest_framework/utils/field_mapping.py:71: AttributeError
```

**Comment by rhblind** [comment]
Hi, thank you for reporting this issue.
I'm trying to keep the dependencies for this library to a minimum in order to avoid dependency issues, so the only requirements are django, django-rest-framework and django-haystack. As long as your versions are within the requirements you should be good. However, I have to admit that this library is unfortunately not getting too much love these days as I'm not currently involved in any projects that uses it. So things might slip by I'm afraid :(

But to me it seems a bit weird that this upgrade causes your error, and I suspect that it's because of the `ManyToOneRel` being used that causes the error. If I take a quick look at the history for the [_get_default_field_kwargs()](https://github.com/inonit/drf-haystack/blame/a14f50b43f4da3fff961d6e8d707f7a8b5013897/drf_haystack/serializers.py#L133) method, it has been calling [get_field_kwargs()](https://github.com/encode/django-rest-framework/blame/9ac9c1b2eafa05cf5b6a605ce54e2d5f2f866f1b/rest_framework/utils/field_mapping.py#L71) from rest framework the whole time, and neither has been changed for a while.

Have this worked before on `ManyToOneRel` fields or is this part of new functionality? That field is a relational field, and I don't think they have support for validators (though I'm not 100% sure).

Basically, we can only serialize fields which has a supported [field mapping](https://github.com/inonit/drf-haystack/blob/a14f50b43f4da3fff961d6e8d707f7a8b5013897/drf_haystack/serializers.py#L91-L111).

I don't know if this was any help, but if you still think this needs to be fixed could you please give me an example for the haystack index which causes the problem?

**Comment by staubina** [comment]
@rhblind Thank you for the quick and highly detailed response!

I will take all that and dig into the issue.  I suspect that the test that is failing here might be the issue as I am not seeing this issue in the code running in our Dev environment with API calls.  So I will dig in and see if there is something in there that is causing the relationship to not work.

The relationship here is not new, but it is possible what you have mentioned.  If I find a solution I will add here and close the issue.

Thank you again.

**Comment by rhblind** [comment]
Perfect, thank you!

**Comment by staubina** [comment]
Resolved this issue by getting the correct combination of upgrades. Hope this helps others in the future.

```
django-haystack==2.8.1
djangorestframework==3.9.3
drf-haystack==1.8.2
```
